src/codeu/chat/server/Controller.java

```diff
@@ -115,6 +122,10 @@ public User newUser(Uuid id, String name, Time creationTime) {
           name,
           creationTime);
 
+      if(!retrieveOn){
```

Comment:
If we receive new messages or users, they're lost if we're writing to the server. Why not save these by adding a timeline event to append the message after the server is done reading the log?